praktikumsplaner-frontend/frontend/src/components/assign/PraktikumsstelleCardZuweisung.vue

```diff
@@ -0,0 +1,261 @@
+            >
+                <v-icon>{{
+                    show ? "mdi-chevron-up" : "mdi-chevron-down"
+                }}</v-icon>
+            </v-btn>
+            <v-expand-transition>
+                <div v-show="show">
+                    <v-divider></v-divider>
+                    <v-card-text>
+                        <p style="white-space: pre-line">
+                            {{ getCardDetailText(props.value) }}
+                        </p>
+                    </v-card-text>
+                </div>
+            </v-expand-transition>
+        </v-card>
+    </v-container>
+</template>
+<script setup lang="ts">
+import { ref } from "vue";
+import Praktikumsstelle from "@/types/Praktikumsstelle";
+import PraktikumsstellenService from "@/api/PraktikumsstellenService";
+import { EventBus } from "@/stores/event-bus";
+import YesNoDialogWithoutActivator from "@/components/common/YesNoDialogWithoutActivator.vue";
+import Nwk from "@/types/Nwk";
+import { findStudiengangColorByValue } from "@/types/Studiengang";
+import { findAusbildungsrichtungColorByValue } from "@/types/Ausbildungsrichtung";
+import { useWarnings } from "@/composables/warningGenerator";
+import { useTextGenerator } from "@/composables/textGenerator";
+
+const props = defineProps<{
+    value: Praktikumsstelle;
+}>();
+
+const show = ref<boolean>(false);
+const unassignDialogContent = ref<string>("");
+const unassignDialogTitle = ref<string>("Zuweisung aufheben?");
+const unassignConfirmDialog = ref<boolean>(false);
+const warningDialog = ref<boolean>(false);
+const nwkToAssignUnassing = ref<Nwk>();
+const warningDialogTitle = ref<string>(
+    "Warnung. Wollen sie wirklich fortfahren?"
+);
+const warningDialogText = ref<string>("");
+const assignedNwk = ref(props.value.assignedNwk);
+const generator = useTextGenerator();
+
+let stelleToAssignUnassign: Praktikumsstelle | undefined;
+
+function getCardText(stelle: Praktikumsstelle): string {
+    return generator.getPraktikumsstellenCardText(stelle);
+}
+
+function getCardDetailText(stelle: Praktikumsstelle): string {
+    return generator.getPraktikumsstellenCardDetailText(stelle);
+}
+
+function drop(event: DragEvent, stelle: Praktikumsstelle) {
+    try {
+        const draggedNwk: Nwk = JSON.parse(
+            event.dataTransfer?.getData("application/json") as string
+        );
+        nwkToAssignUnassing.value = new Nwk(
+            draggedNwk.id,
+            draggedNwk.vorname,
+            draggedNwk.nachname,
+            draggedNwk.jahrgang,
+            draggedNwk.vorlesungstage,
+            draggedNwk.isActive,
+            draggedNwk.studiengang,
+            draggedNwk.ausbildungsrichtung
+        );
+    } catch (e) {
+        return;
+    }
+
+    if (
+        !stelle ||
+        !stelle.id ||
+        stelle.assignedNwk ||
+        nwkToAssignUnassing.value.id === ""
+    ) {
+        return;
+    }
+
+    let warnings = useWarnings().getBeforeAssignmentWarnings(
```

Comment:
In all our files we define useXYXY() vars after imports, and not in methods. This here seems to be an exception which is a bit confusing